plugins/plugin-composer/che-plugin-composer-server/src/main/java/org/eclipse/che/plugin/composer/server/projecttype/ComposerValueProviderFactory.java

```diff
@@ -61,10 +62,8 @@ protected ComposerValueProvider(FolderEntry projectFolder) {
       }
     }
 
-    private JsonObject readModel(FolderEntry projectFolder) throws ServerException, IOException {
-      FileEntry composerFile = (FileEntry) projectFolder.getChild("composer.json");
-      Reader reader = new BufferedReader(new InputStreamReader(composerFile.getInputStream()));
-      return new Gson().fromJson(reader, JsonObject.class);
+    private JsonObject readModel(Path projectFsPath) throws ServerException, IOException {
+      return new Gson().fromJson(Files.newBufferedReader(projectFsPath), JsonObject.class);
```

Comment:
Do we need to close reader here ?

Reply:
I'm not familiar with the logic, I did just source code adaptation to a new API, so I don't really know if we need to close reader here. If it matters for you, I guess you can ask an author.